Kitodo/src/main/java/org/kitodo/production/services/data/ImportService.java

```diff
@@ -964,6 +965,18 @@ private Process loadParentProcess(Ruleset ruleset, int projectId, String parentI
         return parentProcess;
     }
 
+    private List<ProcessDTO> sortProcessesByProjectID(List<ProcessDTO> processDTOs, int projectId) {
```

Comment:
You are adding a sorting here but without to show the project information in the list of possible parent processes - so I don't see any benefit of sorting here.

Reply:
You are right that it does not make sense in case of manual process linking. The origin of that is the idea, that when doing automatic linking during import we have to decide to which parent a child gets linked (if there are multiple). I had the idea of having the parents in the same project to be listed first (and therefor be linked) to not just have a random selection.

Reply:
Do you mean with automatic linking the mass import feature? If yes, as we did not use this feature, I'm not aware of possible issues there. But then your sorting logic is useful and I pray that even then the right parent process is sorted first.

As we do only manual linking with one parent process over all projects we are not effected by this additional sorting and not showing project information. This could / should be added in a different pull request if this information need to be displayed.

Reply:
Yes, this is mostly for mass import. I added an unit test to ensure that sorting works correctly.